src/main/java/com/mixpanel/android/mpmetrics/MixpanelNotificationRouteActivity.java

```diff
@@ -47,47 +64,60 @@ protected void trackAction(Intent routeIntent) {
 
         CharSequence actionIdChars = intentExtras.getCharSequence("actionId");
         if (null == actionIdChars) {
-            MPLog.i(LOGTAG, "Notification action click logged with no actionId.");
+            MPLog.i(LOGTAG, "Notification action click logged with no actionId");
             return;
         }
 
-        CharSequence uriChars = intentExtras.getCharSequence("uri");
-        if (null == uriChars) {
-            MPLog.i(LOGTAG, "Notification action click logged with no uri.");
+        CharSequence actionTypeChars = intentExtras.getCharSequence("actionType");
+        if (null == actionIdChars) {
+            MPLog.i(LOGTAG, "Notification action click logged with no actionType");
+            return;
+        }
+
+        CharSequence labelChars = intentExtras.getCharSequence("label");
+        if (null == labelChars) {
+            MPLog.i(LOGTAG, "Notification action click logged with no label");
             return;
         }
 
         CharSequence messageIdChars = intentExtras.getCharSequence("messageId");
         if (null == messageIdChars) {
-            MPLog.i(LOGTAG, "Notification action click logged with no messageId.");
+            MPLog.i(LOGTAG, "Notification action click logged with no messageId");
             return;
         }
 
         CharSequence campaignIdChars = intentExtras.getCharSequence("campaignId");
         if (null == campaignIdChars) {
-            MPLog.i(LOGTAG, "Notification action click logged with no campaignId.");
+            MPLog.i(LOGTAG, "Notification action click logged with no campaignId");
             return;
         }
 
         final String actionId = actionIdChars.toString();
-        final String uri = uriChars.toString();
+        final String actionType = actionTypeChars.toString();
+        final String label = labelChars.toString();
         final String messageId = messageIdChars.toString();
         final String campaignId = campaignIdChars.toString();
 
         MixpanelAPI.allInstances(new MixpanelAPI.InstanceProcessor() {
             @Override
             public void process(MixpanelAPI api) {
                 JSONObject pushProps = new JSONObject();
-
+                String tapTarget;
                 try {
-                    pushProps.put("actionId", actionId);
-                    pushProps.put("uri", uri);
-                    pushProps.put("messageId", messageId);
-                    pushProps.put("campaignId", campaignId);
+                    if (actionId.equals("notificationClick")) {
```

Comment:
why don't we just set the tapTarget on the intent instead of determining it implicitly? then `actionId` can be null when the target is `notification`